### PR TITLE
Opencl stub fixes for psyclone

### PR DESCRIPTION
**Pull request by hiker**
Some minor changes:
- updated FortCL version in dl_esm_inf
- Updated Makefile to build fortcl in same directory as dl_esm_inf
- Added some dummy variables to grid and field which PSyclone created code accesses, so for PSyclone compile tests to work those variables must be declared.

### CODE REVIEW

File: finite_difference/src/field_mod.f90 [hunk 1]
```diff
@@ -70,6 +70,10 @@ module field_mod
      !> Whether the data for this field lives in a remote memory space
      !! (e.g. on a GPU)
      logical :: data_on_device
+
+     ! TODO: For OpenCL compilation tests in PSyclone we need a device_ptr
```
Comment (arporter) [inline finite_difference/src/field_mod.f90:74]: Please include a reference to the issue number here (#10) and again below.

**Review by arporter — CHANGES_REQUESTED** [review]
I'm not sure what's gone wrong here but attempting to compile this branch fails for me because the `ar` command is still looking for the FortCL object files in the FortCL/src directory. Please could you also bring this branch up to master.

**Comment by hiker** [comment]
Dang, I missed that (I mostly tested in PSyclone, which does not show the message about missing files). Not sure how to fix that - 'ar' all .o files? Or use two separate libraries?

For now I have used the latter - FortCL already creates its own library, so lib_fd will now only contain the dl_esm_inf files (and the application then needs to link with lib_fd and libFortCL). If this is not acceptable, I may have to 'ar' all .o files (since it's not trivial to merge two libraries - see https://stackoverflow.com/questions/3821916/how-to-merge-two-ar-static-libraries-into-one - it either requires gnu specific tools/options, or unpacks the .a files and combines them again, and then we might as well just combine all .o files in the first place).

**Comment by arporter** [comment]
I'd prefer to keep the solution as it is on master (line 102 of src/Makefile) - i.e. compile all the .o files (excluding clfortran.o) into one archive. Does this now become more difficult with the change to build location?

**Comment by hiker** [comment]
It's not more difficult, as long as you are happy that I 'ar' up all .o file (since in dl_esm_inf I don't have easy access to the list of .o files from FortCL). All done

**Comment by arporter** [comment]
That's fine so long as you exclude clfortran.o (or delete it from the archive file afterwards).

**Comment by hiker** [comment]
Yes, I removed that file (though tbh ... does it hurt?? - it's a lot of effort for a few kb of code in a library :grinning: )

**Comment by arporter** [comment]
Oh yes, believe me, it hurts! I spent quite a while debugging my application only to find that I'd linked-in that file and its empty interface definitions were getting called instead of the OpenCL run-time!

**Comment by arporter** [comment]
Compilation works for me now. All previous comments have been addressed.
I've taken the liberty of updating a couple of typos in the comments. No documentation update required.

**Review by arporter — APPROVED** [review]

**Pull request merged**